<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Suleman et al 2024, Genetics underpinnings of fasting and oral glucose-stimulated based insulin sensitivity indices</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="fullcontent">

<div id="quarto-search-results"></div>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Suleman et al 2024, Genetics underpinnings of fasting and oral glucose-stimulated based insulin sensitivity indices</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<!-- ```{r} -->
<!-- #| echo: false -->
<!-- #| include: false -->
<!-- #| warning: false -->
<!-- #| message: false -->
<!-- # Reorder on the IS index  -->
<!-- t2d_all_indices_bmi$is_index <- factor(t2d_all_indices_bmi$is_index, levels = c("1/FIns", "1/HOMA-IR", "Raynaud SI", "QUICKI", "Belfiore basal", "1/FIns/FGlu", "ISI basal", "Bennett SI", "Avignon SI0", "1/Ins 120", "1/Glu 120", "ISI 120", "1/(Ins/Glu)120", "Gutt Index", "Avignon SI120", "Avignon SIM", "Stumvoll Modi", "Stumvoll Dem", "1/IFC", "BIGTT SI", "Matsuda")) -->
<!-- # Reorder Index group -->
<!-- t2d_all_indices_bmi$index_group <- factor(t2d_all_indices_bmi$index_group, levels = c("Fasting", "OGTT_0,120", "OGTT_0,30,120")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #| echo: false -->
<!-- #| include: false -->
<!-- #| warning: false -->
<!-- #| message: false -->
<!-- # Create matrices for the heatmap -->
<!-- # Create a matrix 'snp_index_effect' by selecting 'nearest_gene', 'is_index', and 'alt_effsize' columns -->
<!-- t2d_effect_matx <- t2d_all_indices_bmi %>% -->
<!--   dplyr::select(gene_snp_ra, is_index, alt_effsize) %>%  -->
<!--   tidyr::spread(is_index, alt_effsize) %>%  -->
<!--       as_tibble() %>%  -->
<!--   tibble::column_to_rownames(var = "gene_snp_ra") %>%  -->
<!--   as.matrix() -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # #| echo: false -->
<!-- # #| include: false -->
<!-- # #| warning: false -->
<!-- # #| message: false -->
<!-- #  -->
<!-- # # do clustring  -->
<!-- # set.seed(123) -->
<!-- # m_clusters <- Mclust(t2d_effect_matx) -->
<!-- #  -->
<!-- # # get the cluster classification -->
<!-- # clusters <- m_clusters$classification -->
<!-- # # convert to datafrmae -->
<!-- # data_frame <- as.data.frame(t2d_effect_matx) -->
<!-- # # add cluster column -->
<!-- # data_frame$cluster <- clusters -->
<!-- # # order the rows and then remove the cluster column convert back to matrix -->
<!-- # t2d_effect_mat_ordered <- data_frame[order(data_frame$cluster), -ncol(data_frame)] %>%  -->
<!-- #   as.matrix() -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #| echo: false -->
<!-- #| include: false -->
<!-- #| warning: false -->
<!-- #| message: false -->
<!-- # Assuming t2d_effect_matx is your data matrix -->
<!-- # Calculate the distance matrix for rows and columns -->
<!-- row_dist <- dist(t2d_effect_matx)  # Distance for rows -->
<!-- col_dist <- dist(t(t2d_effect_matx))  # Distance for columns (transpose the matrix) -->
<!-- # Perform hierarchical clustering -->
<!-- row_clust <- hclust(row_dist)  # Clustering for rows -->
<!-- col_clust <- hclust(col_dist)  # Clustering for columns -->
<!-- # Reorder the matrix based on the clustering -->
<!-- t2d_effect_mat_ordered <- t2d_effect_matx[row_clust$order, col_clust$order] -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #| echo: false -->
<!-- #| include: false -->
<!-- #| warning: false -->
<!-- #| message: false -->
<!-- # Create a matrix 'snp_index_effect' by selecting 'nearest_gene', 'is_index', and 'pvalue' columns -->
<!-- t2d_p_mat <- t2d_all_indices_bmi %>% -->
<!--   dplyr::select(gene_snp_ra, is_index, pvalue) %>%  -->
<!--   tidyr::spread(is_index, pvalue) %>%  -->
<!--       as_tibble() %>%  -->
<!--   tibble::column_to_rownames(var = "gene_snp_ra") %>%  -->
<!--   as.matrix() -->
<!-- # ordered the row name sof p_mat to same as the effect_mat -->
<!-- order_names <- rownames(t2d_effect_mat_ordered) -->
<!-- t2d_p_mat_ordred <- t2d_p_mat[order_names, ]  -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #| echo: false -->
<!-- #| include: false -->
<!-- #| warning: false -->
<!-- #| message: false -->
<!-- # Convert the matrix to a long format -->
<!-- effsize_long <- as.data.frame(t2d_effect_mat_ordered) %>% -->
<!--   rownames_to_column(var = "gene_snp_ra") %>% -->
<!--   pivot_longer(cols = -gene_snp_ra, names_to = "is_index", values_to = "alt_effsize") -->
<!-- p_long <- as.data.frame(t2d_p_mat_ordred) %>% -->
<!--   rownames_to_column(var = "gene_snp_ra") %>% -->
<!--   pivot_longer(cols = -gene_snp_ra, names_to = "is_index", values_to = "pvalue") -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #| echo: false -->
<!-- #| include: false -->
<!-- #| warning: false -->
<!-- #| message: false -->
<!-- # Combine the data -->
<!-- combined_data <- effsize_long %>% -->
<!--   left_join(p_long, by = c("gene_snp_ra", "is_index")) %>%  -->
<!--     mutate( -->
<!--     index_group = case_when( -->
<!--       is_index %in% c("1/FIns", "1/HOMA-IR", "Raynaud SI", "QUICKI", "Belfiore basal", "1/FIns/FGlu", "ISI basal", "Bennett SI", "Avignon SI0") ~ "Fasting", -->
<!--       is_index %in% c("1/Ins 120", "1/Glu 120", "ISI 120", "1/(Ins/Glu)120", "Gutt Index", "Avignon SI120", "Avignon SIM", "Stumvoll Modi", "Stumvoll Dem", "1/IFC") ~ "OGTT,0-120", -->
<!--       is_index %in% c("BIGTT SI", "Matsuda") ~ "OGTT,0-30-120", -->
<!--       TRUE ~ "Other"  # For any unmatched case -->
<!--     ) -->
<!--   ) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # #| echo: false -->
<!-- # #| include: false -->
<!-- # #| warning: false -->
<!-- # #| message: false -->
<!-- #  -->
<!-- # combined_data <- combined_data[order(combined_data$alt_effsize, decreasing = TRUE), ] -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #| echo: false -->
<!-- #| include: false -->
<!-- #| warning: false -->
<!-- #| message: false -->
<!-- plotly_heatmap <- plot_ly(data = combined_data, x = ~is_index, y = ~gene_snp_ra, type = "heatmap", z = ~alt_effsize, -->
<!--                           colorscale = list(c(0, "blue"), c(0.5, "white"), c(1, "red")), -->
<!--                           zmin = -0.04, zmax = 0.04, text = ~paste(gene_snp_ra, -->
<!--                                                                    "\nEffect-size:", alt_effsize, -->
<!--                                                                    "\nP-value:", pvalue, -->
<!--                                                                    "\nIndex:", is_index,  -->
<!--                                                                    "\nIndex Group:", index_group), -->
<!--                           hoverinfo = "text") %>% -->
<!--   layout(xaxis = list(title = "Indices", tickangle = 90), -->
<!--          yaxis = list(title = "Gene SNP RA"), -->
<!--          colorbar = list(title = "Beta")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #| echo: false -->
<!-- #| include: false -->
<!-- #| column: screen -->
<!-- #| fig.height: 135 -->
<!-- #| fig-cap: "Insulin sensitiivty indices assocaition with T2D risk alleles" -->
<!-- #| label: Association between 21 insulin sensitivity indices and 426 genetic variants known to be assciated with T2D in population cohort. -->
<!-- plotly_heatmap -->
<!-- ``` -->
<div class="cell page-columns page-full">
<div class="cell-output-display column-screen">
<div id="fig-association-t2d" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="index_files/figure-html/fig-association-t2d-1.png" class="img-fluid figure-img" width="3840"></p>
<figcaption class="figure-caption">Figure&nbsp;1: Insulin sensitivity indices association with T2D risk alleles</figcaption>
</figure>
</div>
</div>
</div>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>